slang.h

```diff
@@ -2909,6 +2913,25 @@ namespace slang
             @return The downstream compiler for that source language */
         virtual SlangPassThrough SLANG_MCALL getDefaultDownstreamCompiler(
             SlangSourceLanguage sourceLanguage) = 0;
+
+            /* Set the 'prelude' placed before generated code for a specific language type.
```

Comment:
I hadn't realized when we discussed `CodeGenTarget` vs `SlangSourceLanguage` that we were talking about public API.

I definitely think it is a bit odd to have this key on a "source language" but then the comment talks about "generated code."

I'm fine with this insofar as it works, but I definitely feel like this API would be less confusing if we dropped the word `Source` from `SlangSourceLanguage` (and/or merged `SlangSourceLanguage` with `CodeGenTarget`).